opm/simulators/wells/MultisegmentWell_impl.hpp

```diff
@@ -837,8 +837,17 @@ namespace Opm
             }
 
             // update the total rate // TODO: should we have a limitation of the total rate change?
-            {
+            {               
                 primary_variables_[seg][GTotal] = old_primary_variables[seg][GTotal] - relaxation_factor * dwells[seg][GTotal];
+
+                // make sure that no injector produce and no producer inject
```

Comment:
Please do not do this, unless we have a good reason to do it.  Similar to in the pressure update, std::max will be likely to be removed. 

Allowing or disallowing injector to produce or producer to inject will be a design problem. Similar to the trick related to function `openCrossFlowAvoidSingularity()` in StandardWell, which was introduced by you to overpass some singularity problem.

Reply:
There is a good reason for avoiding injecting producers since at the moment this is undefined behavior. I don't see any reasons for not introducing this. Please argue why you don't think this is a good solution. It seems to work for the cases I have tested. We already do a lot of chopping, why not here?

Reply:
We might need this feature to test whether a well is operable or not.  The only places that negative rates are dangerous is when THP is involved. I do not have a answer for it yet.

Reply:
Anyway, it is another thing needs to be tested separately if we really want it.

Reply:
I suggest that we merge this then, since it fix a actual and urgent problem for a model 2 realization and you can remove it whenever you have a better solution.

Reply:
> I suggest that we merge this then, since it fix a actual and urgent problem for a model 2 realization and you can remove it whenever you have a better solution.

Please indicate which realization you are talking about for this PR.  Can be done through private message.